.cirrus.yml

```diff
@@ -0,0 +1,559 @@
+freebsd_instance:
+  cpu: 1
+  memory: "1"
+
+install_template: &INSTALL_PLUGIN
+  requirements_script: pkg install -y jq
+  install_script: |
+    release=$(jq -r '.release' $PLUGIN_FILE)
+    name=$(jq '.name' $PLUGIN_FILE)
+    packagesite=$(jq '.packagesite' $PLUGIN_FILE)
+    fingerprints=$(jq -r '.fingerprints | keys[]' $PLUGIN_FILE)
+    plugin_repo=$(jq -r '.artifact' $PLUGIN_FILE)
+    pkgs=$(jq -r '.pkgs | join(" ")' $PLUGIN_FILE)
+
+    pkg_dir=/usr/local/test
+    repos_dir="${pkg_dir}/repos"
+    fingerprints_dir="${pkg_dir}/fingerprints"
+
+    echo "Creating main repos dir: ${repos_dir}"
+    mkdir -p $repos_dir
+    export REPOS_DIR=$repos_dir
+
+    pkg_conf_path="${repos_dir}/test.conf"
+    echo "iocage-plugins: {" > $pkg_conf_path
+    echo "url: $packagesite," >> $pkg_conf_path
+    echo "signature_type: \"fingerprints\"," >> $pkg_conf_path
+    echo "fingerprints \"${fingerprints_dir}\"," >> $pkg_conf_path
+    echo "enabled: true" >> $pkg_conf_path
+    echo } >> $pkg_conf_path
+    echo "Created test pkg config file:"
+    cat $pkg_conf_path
+
+    trusted_fingerprints="$fingerprints_dir/trusted"
+    mkdir -p "${trusted_fingerprints}"
+
+    for repo_name in $fingerprints
+    do
+      repo_fingerprints=$(jq -rc '."fingerprints"."'${repo_name}'"[]' $PLUGIN_FILE)
+
+      repo_count=1
+      echo $repo_fingerprints | while IFS='' read f
+      do
+        echo "Creating fingerprint file for repo:"
+        echo $f
+
+        function=$(echo $f | jq -r '.function')
+        fingerprint=$(echo $f | jq -r '.fingerprint')
+        file_path=${trusted_fingerprints}/${repo_name}_${repo_count}
+
+        echo "Creating new fingerprint file: ${file_path}"
+
+        echo "function: $function" > ${file_path}
+        echo "fingerprint: $fingerprint" >> ${file_path}
+
+        repo_count=$(expr $repo_count + 1)
+      done
+    done
+
+    if [ "$(which git)" = "" ]
+    then
+      pkg install -y git-lite || pkg install -y git
+    fi
+
+    release_branch="$(freebsd-version | cut -d '-' -f1)-RELEASE"
+    git clone -b ${release_branch} ${plugin_repo} ./plugin || git clone -b master ${plugin_repo} ./plugin
+
+    # Cleanup before plugin pkg installation
+    if ! echo ${pkgs} | grep -q "git"
+    then
+      pkg remove -y git-lite git jq
+      pkg autoremove -y
+    fi
+
+    echo "Test install plugin pkgs for plugin: $name"
+    echo "Installing $name pkgs: $pkgs"
+    pkg install -y $pkgs
+
+    if [ -d "./plugin/overlay" ]
+    then
+      echo "Found overlay folder"
+      cp -r ./plugin/overlay/ /
+    fi
+
+    . ./plugin/post_install.sh
+
+    if [ -f ./plugin/pre_update.sh ]
+    then
+      . ./plugin/pre_update.sh
+    fi
+
+    if [ -f ./plugin/post_update.sh ]
+    then
+      . ./plugin/post_update.sh
+    fi
```

Comment:
Just wondering why you decided to source these scripts instead of just running them.

I have another coner case for you - I have _a lot_ of corners, btw :stuck_out_tongue_closed_eyes:

I suppose you could say I cheat, err have some not so typical ways. That is to say, I use `sysrc`... a lot -- So, I have `. /etc/rc.subr && load_rc_config` included with some scripts that may be used during an install or update  -- basically some of my `post_*.sh` scripts are calling other scripts. Here, I have a separate install script because this same script is also used for the upgrades - having the script separate allow it to used during plugin install and dates, and also the provided console menu which allows the service to upgraded at any time, without running a plugin update

When viewing the command output....

- when the post scripts are run directly - [see output](https://cirrus-ci.com/task/5421363269730304?command=install#L162)

![image](https://user-images.githubusercontent.com/35702532/108640257-8d3cf280-7466-11eb-8529-08345385bba6.png)




- when the post scripts are being sourced or `.` - [see output](https://cirrus-ci.com/task/6234190262829056)

![image](https://user-images.githubusercontent.com/35702532/108640205-4ea73800-7466-11eb-92c2-d3df8289edb2.png)

You may notice, in the latter output, the entire `/etc/rc.subr` script is included


It all works the same - again this is a question, not a complaint

Reply:
I was debugging why the homeassistent plugin still failed, but reverted to normal execution instead of just sourcing the files. Will debug some more on another branch, sorry for the confusion. Was thinking there was something wrong during the source of virtualenv inside the homassistant service install script. But realized it is run as another user (using `su` ) entierly so shouldn't matter.

A bit confused why the venv can not be found during the installation right now.

Reply:
> why the homeassistent plugin still failed

`usr/local/etc/rc.d/homeassistant` - Perfect example of my not so typical way's, lol

Reply:
I was looking a lot through this file to understand what was really happening in there :smile: . Haven't had much time to debug homeassitant much more but my latest theory is that the venv module has different version in FreeBSD 12.2 (and 12.2-p2) compared to 12.1 hence the different result.

Reply:
https://cirrus-ci.com/task/6055689542959104 tried to run the failing (ensurepip) command manually in 12.1. Looks like it is the "common" Symbol variable issue. Probably related to e.g. this issue:  https://github.com/ix-plugin-hub/iocage-plugin-index/issues/165. 

(This issue was actually another motivation for this PR, I was thinking about bumping a lot of 12.1 plugins into 12.2 but then I got scared thinking "what if I break something" :stuck_out_tongue:)